pkg/dataobj/internal/dataset/page.go

```diff
@@ -120,10 +120,20 @@ func (p *MemPage) reader(compression datasetmd.CompressionType) (presence io.Rea
 			// directly.
 			zr = zstdPool.New().(*zstdWrapper)
 		}
-
-		return bitmapReader, &closerFunc{Reader: zr, onClose: func() error {
+		defer func() {
 			_ = zr.Reset(nil) // Allow releasing the buffer.
 			zstdPool.Put(zr)
+		}()
+
+		decompressed := bytes.NewBuffer(make([]byte, 0, p.PageInfo().UncompressedSize))
```

Comment:
Do we want to pool these buffers to reduce alloc + GC pressure? It could be released back to the pool in the `onClose`. 

I can help run the full LogQL benchmarks before/after to help figure out the impact of not pooling, if we'd like to see some stats.